### DISCUSSION TITLE
Reuse of bindparams with literal_execute in positional dialects overwrites past values

### DISCUSSION THREAD

**Discussion by tmrlvi** [sqlalchemy/sqlalchemy#10141] [Usage Questions]
When using `bindparam` in `literal_execute`, I get wrong output if I reuse the same bindparam twice.
I suspect it is a bug.
For example,

```
from sqlalchemy import select, create_engine, bindparam

conn = create_engine(f"sqlite://").connect()

a = bindparam("a", value="abc", literal_execute=True)
b = bindparam("b", value="def", literal_execute=True)
c = bindparam("c", value="ghi", literal_execute=True)
print(conn.execute(select(a,b,a,c)).fetchone())
```

**Expected:** ('abc', 'def', 'abc', 'ghi')
**Actual:** ('def', 'def', 'def', 'ghi')

Note that `print(conn.execute(select(a,b,c)).fetchone())` and `print(conn.execute(select(a,a,b,c)).fetchone())` works as expected. The problem is repeating `a` after `b` or `c`. I was able to track it down to `SQLCompiler._process_parameters_for_postcompile` where in `replacement_expressions` the value are overwritten incorrectly, but the root cause is somewhere else.

**Comment by zzzeek** [comment]
thanks for reporting, #10142 is added.
